rootfs/apply.go

```diff
@@ -0,0 +1,136 @@
+package rootfs
+
+import (
+	"io"
+	"io/ioutil"
+
+	"github.com/docker/containerd"
+	"github.com/docker/containerd/log"
+	"github.com/docker/distribution"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+)
+
+type Snapshotter interface {
+	Prepare(key, parent string) ([]containerd.Mount, error)
```

Comment:
> If we force these all to be *just* paths, we will keep the implementations simple and avoid bloat.

Are all the existing graph drivers in Docker able to work with just paths here?  Do any of them need additional metadata about layer contents in order to determine whether a layer already exists during pulls or does the path (via the `key`?) already contain all the necessary information?